do.py

```diff
@@ -4,8 +4,12 @@
 import sys
 import shutil
 import subprocess
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
 
 
+global ixnexception
```

Comment:
is this branch align with main branch? Because I found some changes which is already present in main.